xclim/indices/_threshold.py

```diff
@@ -1003,42 +943,35 @@ def growing_season_start(
     Returns
     -------
     xarray.DataArray, [dimensionless]
-        Day of the year when temperature is superior to a threshold over a given number of days for the first time.
-        If there is no such day or if a growing season is not detected, returns np.nan.
-
-    Notes
-    -----
-    Let :math:`x_i` be the daily mean temperature at day of the year :math:`i` for values of :math:`i` going from 1
-    to 365 or 366. The start date of the start of growing season is given by the smallest index :math:`i`:
-
-    .. math::
-
-       \prod_{j=i}^{i+w} [x_j >= thresh]
-
-    where :math:`w` is the number of days the temperature threshold should be met or exceeded,
-    and :math:`[P]` is 1 if :math:`P` is true, and 0 if false.
+        Start of the growing season.
     """
-    thresh = convert_units_to(thresh, tas)
-    cond = compare(tas, op, thresh, constrain=(">=", ">"))
-
-    out = cond.resample(time=freq).map(rl.first_run, window=window, coord="dayofyear")
-    out.attrs.update(units="", is_dayofyear=np.int32(1), calendar=get_calendar(tas))
-    return out
+    return season(
+        tas,
+        thresh=thresh,
+        mid_date=mid_date,
+        window=window,
+        freq=freq,
+        op=op,
+        constrain=(">", ">="),
+        stat="start",
+    )
 
 
 @declare_units(tas="[temperature]", thresh="[temperature]")
 def growing_season_end(
     tas: xarray.DataArray,
     thresh: Quantified = "5.0 degC",
-    mid_date: DayOfYearStr = "07-01",
+    mid_date: DayOfYearStr | None = "07-01",
```

Comment:
If this is an Optional field, it should be equal to `None` by default. Perhaps mention in the docstring that this can be set to `None`?

Reply:
I can't say for the type that would be better for this job, but I think that the logic is:

* we want the default behaviour that the season includes july 1rst
* but another possibility is to put no such constraint

So we don't want to use `None` as default? Perhaps calling it optional in this case is a bit weird, but there should be a way to allow this behaviour?

Reply:
I suppose we can leave it as is; The warning about Northern Hemisphere locations is there. So long as the value `None` can be passed here, it's valid.